### PR TITLE
Levana eth airdrop contract [WIP]

### PR DESCRIPTION
**Pull request by humanalgorithm**
Rebranching bc other branch diverged from main

### CODE REVIEW

File: contracts/sg-eth-airdrop/src/integration_tests.rs [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+pub fn contract() -> Box<dyn Contract<StargazeMsgWrapper>> {
+    let contract = ContractWrapper::new(
+        crate::contract::execute,
+        crate::contract::instantiate,
+        crate::contract::query,
+    );
+    Box::new(contract)
+}
+
+fn get_instantiate_contract<'a>(claim_plaintext: &str) -> StargazeApp {
```
Comment (yubrew) [inline contracts/sg-eth-airdrop/src/integration_tests.rs:32]: why is there a lifetime here? seems unnecessary

File: contracts/sg-eth-airdrop/src/integration_tests.rs [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    let mut app = custom_mock_app();
+    let true_admin = Addr::unchecked("true_admin");
```
Comment (yubrew) [inline contracts/sg-eth-airdrop/src/integration_tests.rs:34]: probably better to define "true_admin" as a constant

File: contracts/sg-eth-airdrop/src/ethereum.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+pub fn decode_address(input: &str) -> StdResult<[u8; 20]> {
+    if input.len() != 42 {
+        return Err(StdError::generic_err(
+            "Ethereum address must be 42 characters long",
+        ));
+    }
+    if !input.starts_with("0x") {
+        return Err(StdError::generic_err("Ethereum address must start wit 0x"));
```
Comment (yubrew) [inline contracts/sg-eth-airdrop/src/ethereum.rs:45]: with typo

**Review by yubrew — COMMENTED** [review]
a few comments for now before digging in

File: contracts/sg-eth-airdrop/src/integration_tests.rs (hunk 2 above)
Comment (humanalgorithm) [reply contracts/sg-eth-airdrop/src/integration_tests.rs]: Now using OWNER for admin calls

File: contracts/sg-eth-airdrop/src/integration_tests.rs (hunk 1 above)
Comment (humanalgorithm) [reply contracts/sg-eth-airdrop/src/integration_tests.rs]: Removed

File: contracts/sg-eth-airdrop/src/ethereum.rs (hunk 3 above)
Comment (humanalgorithm) [reply contracts/sg-eth-airdrop/src/ethereum.rs]: Fixed

**Pull request closed without merging**